src/libstore/misc.cc

```diff
@@ -301,4 +301,49 @@ std::map<DrvOutput, StorePath> drvOutputReferences(
     return drvOutputReferences(Realisation::closure(store, inputRealisations), info->references);
 }
 
+OutputPathMap resolveDerivedPath(Store & store, const DerivedPath::Built & bfd, Store * evalStore0)
```

Comment:
Maybe rename `evalStore0` to `evalStore_`. That's what we usually do for shadowed names.

Reply:
Actually why does `evalStore` need to be passed as a pointer instead of a reference?

Reply:
In the header it's has a default argument: `evalStore = nullptr`. I would like to do `evalStore = *this` and make it a reference but I don't think that is valid C++?